modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryParser.java

```diff
@@ -134,6 +135,49 @@ public QueryParserResult parse(String schemaName, SqlFieldsQuery qry, boolean re
         return res;
     }
 
+    /**
+     * Create parameters from query.
+     *
+     * @param qry Query.
+     * @return Parameters.
+     */
+    public QueryParameters fromQuery(SqlFieldsQuery qry) {
+        NestedTxMode nestedTxMode = NestedTxMode.DEFAULT;
+        boolean autoCommit = true;
+        List<Object[]> batchedArgs = null;
+
+        if (qry instanceof SqlFieldsQueryEx) {
+            SqlFieldsQueryEx qry0 = (SqlFieldsQueryEx)qry;
+
+            if (qry0.getNestedTxMode() != null)
+                nestedTxMode = qry0.getNestedTxMode();
+
+            autoCommit = qry0.isAutoCommit();
+
+            batchedArgs = qry0.batchedArguments();
+        }
+
+        int timeout = 0;
+
+        if (qry.getTimeout() > 0)
+            timeout = qry.getTimeout();
+        else
+            timeout = (int)idx.kernalContext().config().getDefaultQueryTimeout();
```

Comment:
Actually, numeric overflow problems are interesting here. It seems that max Integer value is not so big, do we validate a value when use `setDefaultQueryTimeout`?

Reply:
yes, we validate timeout in `QueryUtils.validateTimeout` which is invoked inside `SqlFieldsQuery.setTimeout`.

I think `int` is mainly used since we check remaining timeout in IgniteH2Indexing.operationTimeout using Math.min/Math.max methods